### PR TITLE
Golang map is not thread-safe. So sometime this library is panic-ing.

### PR DESCRIPTION
**Pull request by govindarajan**
- Adding lock whenever there is read/write operation happening on map.

[Review on Reviewable](https://reviewable.io/reviews/ttacon/libphonenumber/25)

### CODE REVIEW

File: phonenumberutil.go [hunk 1]
```diff
@@ -549,25 +556,127 @@ var (
...
+	ccForNGRMutex                        sync.RWMutex
 )
 
 var ErrEmptyMetadata = errors.New("empty metadata")
 
+func readFromRegexCache(key string) (*regexp.Regexp, bool) {
+	regCacheMutex.RLock()
+	v, ok := regexCache[key]
+	regCacheMutex.RUnlock()
+	runtime.Gosched()
```
Comment (ttacon) [inline phonenumberutil.go]: Could we remove all the calls to Gosched? I don't think it's a library's job to try to directly interact with the scheduler if it doesn't specifically say that it's going to.

**Comment by ttacon** [comment]
Thanks for the PR! Could we actually remove the usage of runtime.Gosched() though?

**Comment by ttacon** [comment]
Also, we only need to wrap the `regexCache` map in a lock, all the other maps are only ever written to once in the `init()` func which won't need to worry about synchronizing map writes/reads.

**Comment by loopfz** [comment]
Any news?

**Comment by steeve** [comment]
maybe you should squash the commits into one ?

**Comment by brianestlin** [comment]
@ttacon @govindarajan Is this PR ready to be merged? I would like to use libphonenumber in a production server and it would be nice if it didn't panic. ;)

**Pull request merged**